libnavigation-core/src/main/java/com/mapbox/navigation/core/reroute/MapboxRerouteController.kt

```diff
@@ -0,0 +1,98 @@
+package com.mapbox.navigation.core.reroute
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.base.common.logger.Logger
+import com.mapbox.base.common.logger.model.Message
+import com.mapbox.base.common.logger.model.Tag
+import com.mapbox.navigation.core.directions.session.DirectionsSession
+import com.mapbox.navigation.core.directions.session.RoutesRequestCallback
+import com.mapbox.navigation.core.routeoptions.RouteOptionsProvider
+import com.mapbox.navigation.core.trip.session.TripSession
+
+/**
+ * Default implementation of [RerouteController]
+ */
+internal class MapboxRerouteController(
+    private val directionsSession: DirectionsSession,
+    private val tripSession: TripSession,
+    private val routeOptionsProvider: RouteOptionsProvider,
+    private val logger: Logger
+) : RerouteController {
+
+    internal companion object {
+        const val TAG = "MapboxRerouteController"
+    }
+
+    override var rerouteState: RerouteState = RerouteState.IDLE
+        set(value) {
+            if (field == value) {
+                return
+            }
+            field = value
+        }
+
+    // current implementation ignore `onNewRoutes` callback because `DirectionsSession` update routes internally
+    override fun reroute(routesCallback: RerouteController.RoutesCallback) {
```

Comment:
Let's call `rerouteState = RerouteState.FETCHING_ROUTE` when this method is called, not when `tryReroute` is called.

Reply:
yeah, like it